### PR TITLE
Build docs with sphinx-gallery

### PR DESCRIPTION
**Pull request by flothesof**
Hi all,

this is a first draft of sphinx-gallery generation of examples.
Let me know what you think of it.

Regards,
Florian

### CODE REVIEW

**Comment by flothesof** [comment]
I managed to get it to work, despite the windows specific problems. Among the fixes are things like this:
```
-            lsnvert = np.asarray(lsnvert, dtype=np.int64)
+            lsnvert = np.asarray(lsnvert, dtype=np.long)
```
There are probably some more of these to be found: they cause problems on Windows!

File: tofu/geom/_core.py [hunk 1]
```diff
@@ -3665,7 +3665,7 @@ def _prepare_inputs_kInOut(self, D=None, u=None, indStruct=None):
                 else:
                     num_tot_structs += len(ss.Lim)
 
-            lsnvert = np.asarray(lsnvert, dtype=np.int64)
+            lsnvert = np.asarray(lsnvert, dtype=np.long)
```
Comment (Didou09) [inline tofu/geom/_core.py]: Not sure how this affects linux users, waiting for feedback from Laura

File: tofu/geom/utils.py [hunk 2]
```diff
@@ -35,8 +36,8 @@
 _lok = np.array([_lok, _lok+10])
 
 _here = os.path.abspath(os.path.dirname(__file__))
-_root = _here[:_here.rfind('/tofu')]
-_path_testcases = os.path.join(_root,'tofu/geom/inputs')
+_root = tofu.__path__[0]
+_path_testcases = os.path.join(_root, 'geom', 'inputs')
```
Comment (Didou09) [inline tofu/geom/utils.py]: good practice, thanks

File: tofu/geom/_core.py (hunk 1 above)
Comment (lasofivec) [reply tofu/geom/_core.py]: I'm gonna test it in my machine, could you try it also in CentOS ?

**Comment by lasofivec** [comment]
Made changes for pep8, merged with `devel`, and tried on my Mac and Ubuntu. Should we wait for @Didou09 to try on CentOS?

**Comment by lasofivec** [comment]
Forgot to mention: the example is 🔥

**Pull request merged**